scene/resources/surface_tool.cpp

```diff
@@ -729,6 +729,21 @@ Ref<ArrayMesh> SurfaceTool::commit(const Ref<ArrayMesh> &p_existing, uint64_t p_
 
 	Array a = commit_to_arrays();
 
+	// Disable compression if all z euqals 0 (the mesh is 2D)
+	Vector<Vector3> mesh_vertices = a[RS::ARRAY_VERTEX];
+	bool is_mesh_2d = true;
+
+	for (int i = 0; i < mesh_vertices.size(); i++) {
+		if (!Math::is_zero_approx(mesh_vertices[i].z)) {
+			is_mesh_2d = false;
+			break;
+		}
+	}
+
+	if (is_mesh_2d) {
+		p_compress_flags = 0;
```

Comment:
Should this code path set `ARRAY_FLAG_USE_2D_VERTICES` ?
Also, what if there are 3D normals or tangents whose z value is nonzero? There are cases in 3D where a mesh will have zero z, for example exporting a plane from Blender for use as a ground surface.

Reply:
> There are cases in 3D where a mesh will have zero z

Yes I do agree there should be more discussions about how to check whether the mesh is 2D or not, this pr is just a hijack.  But on the other hand, IMHO disable mesh compression for the case you mentioned is not a severe problem.